packages/playwright-test/src/loader.ts

```diff
@@ -143,7 +143,19 @@ export class Loader {
     this._fullConfig.shard = takeFirst(config.shard, baseFullConfig.shard);
     this._fullConfig._ignoreSnapshots = takeFirst(config.ignoreSnapshots, baseFullConfig._ignoreSnapshots);
     this._fullConfig.updateSnapshots = takeFirst(config.updateSnapshots, baseFullConfig.updateSnapshots);
-    this._fullConfig.workers = takeFirst(config.workers, baseFullConfig.workers);
+
+    const workers = takeFirst(config.workers, baseFullConfig.workers);
```

Comment:
Let's make `baseFullConfig.workers` be `50%`, remove cpu logic from there and only leave it here. Note that today it has some special considerations for M1 that we decided to remove.

Reply:
Done.

Reply:
I set the `50%` on this line instead of `baseFullConfig`, to don't mess up with `FullConfigInternal.workers` typing for no reason. I believe it is not useful to have `FullConfigInternal.workers` as `number|string`  since it actually represent the number of workers internally.

What do you think ?